tasks_week3_Shai_Axelrod/Zoo/include/Zoo.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+#include "Animal.h"
+#include <vector>
+
+#define ROWS 20
+#define COLUMNS 40
+enum AnimalType {
+#define ANIMAL(a) E_##a
+#include "AnimalTypes.h"
```

Comment:
what is that ??

Reply:
it is a common and nice technique. that utilize the preprocessor commands.
here it enables us to add supported animal type in a dedicated file (animalTypes.h) and they will added automatically to the supported type list as well as the supported types enum